### ISSUE TITLE
Better Error Reporting / Eliminated "Silent Fails"

### ISSUE THREAD

**Issue by MelanieTolomeo** [SFDO-Community/declarative-lookup-rollup-summaries#1243]
**Is your feature request related to a problem? Please describe.**
This is from one of my colleagues in proserv/Traction:

> when an established rollup doesn't work (for whatever reason... bad data, unavailable records, something something) there's no kind of notification or warning that you need to take a look at it.  As a result, your reporting can be wrong for a really long time until someone thinks to take a deeper look into it and analyse the numbers themselves...

**Describe the solution you'd like**

>   If we can get a notification of some kind, error log generated, something - then we could build a process around that and get someone looking into an issue...

Notification can have:

- Rollup name/ID
- Parent record
- Child record(s)
- Issue description (in plain English)
- Date

**Describe alternatives you've considered**
- Error logs exist in scheduled incremental run mode but no others. This suggested error handling would extend to all run modes, including full schedule, realtime, Apex, etc.
- Currently need to create manual exception reports and see where the discrepancies lie, somewhat defeating the purpose of the managed package and causing data trust issues

**Additional context**
- Idea for how to fix: [Error handling API for NPSP](https://github.com/SalesforceFoundation/NPSP/blob/main/force-app/main/default/classes/ERR_Handler_API.cls)
- Colleague said he had a problem with a consulting client where a lot of rework was needed because there were DLRS discrepancies and no  way to see where they were. He currently does not recommend the tool to clients.

**Comment by aheber** [comment]
@MelanieTolomeo do you have details about what kind of silent fails are occurring? What causes errors so we can narrow in on them?

Right now this issue is too vague to be actionable.

**Comment by aheber** [comment]
Unable to work on this issue. Future issues should be specific about problems being experienced.
